### PR TITLE
add CASSANDRA_DELAY_START_TIME environment variable

### LINKED ISSUE #55317: [bitnami/cassandra] Add delayed start of Cassandra start/setup
### Name and Version

bitnami/cassandra:4.1

### What is the problem this feature will solve?

We want to run Bitnami Cassandra in AWS ECS Cluster with AWS Cloud Map DNS integration.
This setup does the following:
1. AWS ECS service creates task with Cassandra container and assigns it internal IP address by attaching AWS ENI
2. AWS Cloud Map registers newly create task in AWS Route 53 DNS with the assigned IP address

The problem is that it doesn't happen instantly. There is nearly 10-60 seconds delay between step 1 and step 2.
Unfortunately, that is enough for Cassandra container to start and find that its hostname can't be resolved and if it is SEED node it will fail with:
```
WARN  [main] 2024-01-23 16:34:26,326 SimpleSeedProvider.java:62 - Seed provider couldn't lookup host seed1.domain.com
WARN  [main] 2024-01-23 16:34:26,330 SimpleSeedProvider.java:62 - Seed provider couldn't lookup host seed2.domain.com
ERROR [main] 2024-01-23 16:34:26,332 CassandraDaemon.java:897 - Exception encountered during startup: The seed provider lists no seeds.
```

### What is the feature you are proposing to solve the problem?

I propose to add an environment variable `CASSANDRA_DELAY_START_TIME`. That will pause Cassandra start on the number of seconds specified in that variable.
This delay should be implemented before the Cassandra process starts.
I can see two places where this can be implemented. I have tested both of them and they do work properly:
1. Implement sleep delay directly in the entrypoint.sh. That will make sure that no scripts will start before our delay passes.
2. Implement sleep delay in function [cassandra_start_bg](https://github.com/bitnami/containers/blob/af0b58b8fddad0c4886ff09b39cd83a4f0fbdcbf/bitnami/cassandra/4.1/debian-11/rootfs/opt/bitnami/scripts/libcassandra.sh#L994). This also works, but part of seutp.sh script already executed before this step.

I prefer the first option with sleep delay directly in the entrypoint.sh

### What alternatives have you considered?

I have tried the following solutions:
1. Rewrite CMD - that doesn't work because entrypoint.sh starts execution of setup.sh before CMD is executed.
2. Variable CASSANDRA_INIT_SLEEP_TIME - doesn't work because it is used after the Cassandra process has already been started
3. Custom init script - doesn't work because init scripts executed after the Cassandra process has already been started

### PR DESCRIPTION
**Pull request by dpavlov-smartling**
### Description of the change

This is PR for issue https://github.com/bitnami/containers/issues/55317
It adds environment variable `CASSANDRA_DELAY_START_TIME` that delays Cassandra start by the number of provided seconds.

### Benefits

This adds possibility to delay Cassandra start and give DNS service time to register newly created container/task. This helps if you use AWS ECS + AWS Cloud Map
### Possible drawbacks
TBD
### Applicable issues

- fixes https://github.com/bitnami/containers/issues/55317

### CODE REVIEW

File: bitnami/cassandra/4.0/debian-11/rootfs/opt/bitnami/scripts/cassandra/entrypoint.sh [hunk 1]
```diff
@@ -18,6 +18,11 @@ set -o pipefail
 
 print_welcome_page
 
+if is_positive_int $CASSANDRA_DELAY_START_TIME && [ $CASSANDRA_DELAY_START_TIME -gt 0 ]; then
+    info "** Delaying Cassandra start by $CASSANDRA_DELAY_START_TIME seconds **"
+    sleep $CASSANDRA_DELAY_START_TIME
+fi
+
```
Comment (fmulero) [inline bitnami/cassandra/4.0/debian-11/rootfs/opt/bitnami/scripts/cassandra/entrypoint.sh:25]: A little suggestion to match with our code styles.

```suggestion
if is_positive_int "$CASSANDRA_DELAY_START_TIME" && [[ "$CASSANDRA_DELAY_START_TIME" -gt 0 ]]; then
    info "** Delaying Cassandra start by ${CASSANDRA_DELAY_START_TIME} seconds **"
    sleep "$CASSANDRA_DELAY_START_TIME"
fi

```

File: bitnami/cassandra/4.1/debian-11/rootfs/opt/bitnami/scripts/cassandra/entrypoint.sh [hunk 2]
```diff
@@ -18,6 +18,11 @@ set -o pipefail
 
 print_welcome_page
 
+if is_positive_int $CASSANDRA_DELAY_START_TIME && [ $CASSANDRA_DELAY_START_TIME -gt 0 ]; then
+    info "** Delaying Cassandra start by $CASSANDRA_DELAY_START_TIME seconds **"
+    sleep $CASSANDRA_DELAY_START_TIME
+fi
+
```
Comment (fmulero) [inline bitnami/cassandra/4.1/debian-11/rootfs/opt/bitnami/scripts/cassandra/entrypoint.sh:25]: A little suggestion to match with our code styles.

```suggestion
if is_positive_int "$CASSANDRA_DELAY_START_TIME" && [[ "$CASSANDRA_DELAY_START_TIME" -gt 0 ]]; then
    info "** Delaying Cassandra start by ${CASSANDRA_DELAY_START_TIME} seconds **"
    sleep "$CASSANDRA_DELAY_START_TIME"
fi

```

**Review by fmulero — COMMENTED** [review]
Hi @dpavlov-smartling Thanks a lot for your contribution. I hope you are doing well.

I left a little suggestion, Could your give it a glance?

**Comment by dpavlov-smartling** [comment]
Applied suggestions

**Review by fmulero — APPROVED** [review]
LGTM

**Pull request merged**